src/factory.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * factory.ts
+ * Generate a factory graph using a recursive algorithm.
+ * lgfrbcsgo & Nikolaus - October 2020
+ */
+
+import { items, Item, Ore, isOre } from "./items"
+import { findRecipe } from "./recipes"
+import { ContainerNode, FactoryGraph, IndustryNode, PerMinute } from "./graph"
+
+/**
+ * Add to the a factory graph all nodes required to produce and store a given item
+ * produced at a given rate. Recursively call this function to produce all ingredients.
+ * @param factory FactoryGraph to be modified
+ * @param item Item to be built
+ * @param rate Rate at which to build the product
+ */
+export function buildDependencies(
+    factory: FactoryGraph,
+    item: any,
```

Comment:
Please don't use `any` if possible. `any` introduces unsoundness into your types.
```suggestion
    item: Item,
```
If we use `Item` instead, we need to prove that item is not an ore when calling `findRecipe`.

Reply:
Fixed in https://github.com/lgfrbcsgo/du-factory-generator/commit/b2ca5abf70d5f5d1f8272961ae611d9e8b5b9bf3